infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -294,9 +295,15 @@ private void rangeWrite(CorfuPayloadMsg<RangeWriteMsg> msg,
      * Warning: Clears all data.
      */
     @ServerHandler(type = CorfuMsgType.RESET_LOGUNIT)
-    private void resetLogUnit(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    private synchronized void resetLogUnit(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        // Waits for all current writes to get flushed
+        // This method is called after a seal and before consensus
+        // therefore writes wouldn't interleave with the reset
+        log.info("resetLogUnit: reset server msg1 {}", msg);
         streamLog.reset();
         dataCache.invalidateAll();
+        batchWriter = new BatchWriter(streamLog);
```

Comment:
Can't we pass the flush and reset through the batchwriter as it ensures single threaded execution.

Reply:
this logic will be called after a seal, so I don't think you can have interleaving if you clear/wait for the batchwriter to finish all its work then do the reset.